src/lib/bootloader/grub2pwd.rb

```diff
@@ -1,63 +1,131 @@
 require "yast"
 
-# class is responsible for detection, encryption and writing of grub2 password protection
-class GRUB2Pwd
-  YAST_BASH_PATH = Yast::Path.new(".target.bash_output")
-  PWD_ENCRYPTION_FILE = "/etc/grub.d/42_password"
+Yast.import "Stage"
 
-  def used?
-    Yast.import "FileUtils"
+module Bootloader
+  # class is responsible for detection, encryption and writing of grub2 password protection
+  class GRUB2Pwd
+    attr_accessor :used, :unrestricted
+    alias_method :used?, :used
+    alias_method :unrestricted?, :unrestricted
 
-    Yast::FileUtils.Exists PWD_ENCRYPTION_FILE
-  end
+    def initialize
+      # TODO: offline upgrade should somehow read it when we need grub2 -> grub2 modifications
+      if Yast::Stage.initial
+        propose
+      else
+        read
+      end
+    end
 
-  def enable(password)
-    enc_passwd = encrypt(password)
-
-    file_content = "#! /bin/sh\n" \
-      "exec tail -n +3 $0\n" \
-      "# File created by YaST and next password change in YaST will overwrite it\n" \
-      "set superusers=\"root\"\n" \
-      "password_pbkdf2 root #{enc_passwd}\n" \
-      "export superusers"
-
-    Yast::SCR.Write(
-      Yast::Path.new(".target.string"),
-      [PWD_ENCRYPTION_FILE, 0700],
-      file_content
-    )
-  end
+    def write
+      if used?
+        enable
+      else
+        disable
+      end
+    end
 
-  def disable
-    return unless used?
+    def password=(value)
```

Comment:
So the state "password is set but disabled" can exist, but only in the UI, not in the config files, right?